lib/model/compose.dart

```diff
@@ -175,3 +176,58 @@ Uri narrowLink(PerAccountStore store, Narrow narrow, {int? nearMessageId}) {
 
   return store.account.realmUrl.replace(fragment: fragment.toString());
 }
+
+String mention(User user, {bool silent = false}) {
+  return '@${silent ? '_' : ''}**${user.fullName}|${user.userId}**';
+}
+
+/// https://spec.commonmark.org/0.30/#inline-link
+///
+/// The "link text" is made by enclosing [visibleText] in square brackets.
+/// If [visibleText] has unexpected features, such as square brackets, the
+/// result may be surprising.
+///
+/// The part between "(" and ")" is just a "link destination" (no "link title").
+/// That destination is simply the stringified [destination], if provided.
+/// If that has parentheses in it, the result may be surprising.
+// TODO (?) try harder to guarantee output that creates an inline link,
+//   and in particular, the intended one. Tricky in part because that depends on
```

Comment:
I believe it does have a way that works and is [supported by the Markdown spec](https://spec.commonmark.org/0.30/#entity-and-numeric-character-references): for example, replace `]` with `&#x5D;`. But I think that doesn't solve the issue you linked to, which is specifically about backslash escaping.

Reply:
Hmm interesting!

I think that is likely to work. Probably this should still be left as a TODO — it's complicated somewhat by wanting to avoid turning *everything* into such references, so that typical text goes through verbatim. But it'd be good for the TODO to mention that idea.

That issue's title is about backslash escaping, but I think a lot of the demand for it is really just about *some* form of escaping that a piece of software can use when generating Zulip messages. Sadly it looks like the HTML references (and quite possibly backslashes too) won't solve that problem for all use cases, as I [just found by experiment](https://chat.zulip.org/#narrow/stream/9-issues/topic/broken.20topic.20link/near/1596986) — in particular it doesn't work properly inside our topic-mention syntax, and perhaps not in some of our other custom Markdown features. But it does seem to work in our implementation inside ordinary Markdown links, like we're discussing here.